torch/_dynamo/utils.py

```diff
@@ -913,6 +919,61 @@ def enum_repr(value, local):
     return local_name
 
 
+def is_hashable_python_var(x):
+    from .allowed_functions import is_builtin_callable
+    from .variables import ConstantVariable
+
+    return (
+        ConstantVariable.is_literal(x)
+        or isinstance(x, torch.Tensor)
+        or is_builtin_callable(x)
+    )
+
+
+def is_hashable(x):
+    from .variables import (
+        BuiltinVariable,
+        ConstantVariable,
+        EnumVariable,
+        TensorVariable,
+    )
+
+    return isinstance(
+        x,
+        (
+            BuiltinVariable,
+            ConstantVariable,
+            EnumVariable,
+            TensorVariable,
+        ),
+    )
+
+
+class HashableTracker:
+    def __init__(self, vt):
+        assert is_hashable(vt), type(vt)
+        self.vt = vt
+
+    @property
+    def underlying_value(self):
+        from .variables import TensorVariable
+
+        if isinstance(self.vt, TensorVariable):
+            x = self.vt.as_proxy().node
```

Comment:
Consider using `node.meta['example_value']` instead.

Reply:
What if there is no example value?

Reply:
Turns out the right thing to use is `self.vt.source`

Reply:
Technically, every TensorVariable ought to have an example value - tensor inputs to dynamo are all wrapped, while ideally every tensor variable is produced as a result of running wrap_proxy_call.

I’m not sure if this is true if you create tensor variables from symnodes though…

Reply:
Not sure what's the current state, but if you create a tensor from a symnode you could very well use the example value within the symnode. If it's unbacked, it means that you could use any valid value, I guess.

Reply:
Hmm, is the example value of the tensor constructed from symnode result in a FakeTensor? I guess it would. Let me bring up the code for it.